books_scraper/spiders/books_spider.py

```diff
@@ -0,0 +1,28 @@
+import scrapy
+from scrapy.http import Response
+
+
+class BooksSpiderSpider(scrapy.Spider):
+    name = "books_spider"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    def parse(self, response: Response, **kwargs):
+        for book in response.css("article.product_pod"):
+            detail_page_url = book.css("h3 a::attr(href)").get()
+            yield response.follow(detail_page_url, callback=self.parse_book_details)
```

Comment:
Ensure that the `detail_page_url` is correctly constructed. The current implementation assumes that the URL is relative and will be correctly resolved by `response.follow`. If there are any issues with URL resolution, consider using `response.urljoin`.